src/billing/components/Checkout/BillingContactForm.tsx

```diff
@@ -0,0 +1,343 @@
+import React, {FC, useState} from 'react'
+import {
+  Form,
+  Input,
+  Columns,
+  SelectDropdown,
+  ComponentColor,
+  Button,
+  ComponentStatus,
+  Grid,
+  Panel,
+  ComponentSize,
+  FlexBox,
+  JustifyContent,
+  Alert,
+  IconFont,
+} from '@influxdata/clockface'
+
+import BillingContactSubdivision from 'src/billing/components/Checkout/BillingContactSudivision'
+import {states, countries} from 'src/billing/constants'
+
+import 'babel-polyfill'
+import axios from 'axios'
+import {convertKeysToSnakecase} from 'src/billing/utils/checkout'
+import {useBilling} from 'src/billing/components/BillingPage'
+
+type Props = {
+  onSubmitForm: () => void
+}
+
+const BillingContactForm: FC<Props> = ({onSubmitForm}) => {
+  const [
+    {
+      account: {billingContact},
+    },
+  ] = useBilling()
+
+  const [firstName, setFirstName] = useState(billingContact.firstName)
+  const [lastName, setLastName] = useState(billingContact.lastName)
+  const [companyName, setCompanyName] = useState(billingContact.companyName)
+  const [country, setCountry] = useState(billingContact.country)
+  const [street1, setStreet1] = useState(billingContact.street1)
+  const [street2, setStreet2] = useState(billingContact.street2)
+  const [city, setCity] = useState(billingContact.city)
+  const [subdivision, setSubdivision] = useState(billingContact.subdivision)
+  const [postalCode, setPostalCode] = useState(billingContact.postalCode)
+
+  const [firstNameError, setFirstNameError] = useState(false)
+  const [lastNameError, setLastNameError] = useState(false)
+  const [companyNameError, setCompanyNameError] = useState(false)
+  const [countryError, setCountryError] = useState(false)
+  const [cityError, setCityError] = useState(false)
+  const [subdivisionError, setSubdivisionError] = useState(false)
+  const [isSubmittingContact, setIsSubmittingContact] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const requiredErrorText = 'This is a required field'
+
+  const handleSetFirstName = (e): void => {
+    const {value} = e.target
+    setFirstNameError(value.trim() === '')
+    setFirstName(value)
+  }
+
+  const handleSetLastName = (e): void => {
+    const {value} = e.target
+    setLastNameError(value.trim() === '')
+    setLastName(value)
+  }
+
+  const handleCityChange = (e): void => {
+    const {value} = e.target
+    setCityError(value.trim() === '')
+    setCity(value)
+  }
+
+  const handleStreet1Change = (e): void => {
+    const {value} = e.target
+    setStreet1(value)
+  }
+
+  const handleStreet2Change = (e): void => {
+    const {value} = e.target
+    setStreet2(value)
+  }
+
+  const handlePostalCodeChange = (e): void => {
+    const {value} = e.target
+    setPostalCode(value)
+  }
```

Comment:
Rather than have `n` handlers that can handle each input separately you could use this neat little trick using the input `name` attribute:

```
  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    const key = e.target.name // name can be "street1", "postalCode", "company" etc.
    const inputState = {...allInputs, [key]: value}

    setInputs(inputState)
  }
```